src/architecture/principles.md

```diff
@@ -0,0 +1,287 @@
+environments, and in how intelligence can be applied at the endpoints in novel ways, while
+being strict with outcomes guarantees that the result will be correct and interoperable.
+
+Note that this approach to robustness also covers the
+[End-to-end Principle](https://en.wikipedia.org/wiki/End-to-end_principle). The end-to-end
+principle states that the reliability properties of a protocol have to be
+supported at its endpoints and not in intermediary nodes. For instance, you can best guarantee
+the confidentiality or authenticity of a message by encrypting or signing at one endpoint and
+decrypting or verifying at the other rather than asking relaying nodes to implement local
+protections. IPFS's aproach to robustness, via CIDs, is well aligned with that principle.
+
+## IPFS Implementation Requirements
+
+An :dfn[IPFS Implementation]:
+- MUST support addressability using CIDs.
+- MUST expose operations (eg. retrieval, provision, indexing) on resources using CIDs. The operations
+  that an implementation may support is an open-ended set, but this requirement should cover any interaction
+  which the implementation exposes to other IPFS implementations.
+- MUST verify that the CIDs it resolves match the resources they address, at least when it
+  has access to the resources' bytes. Implementations MAY relax this requirement in
+  controlled environments in which it is possible to ascertain that verification has happened
+  elsewhere in a trusted part of the system.
+- SHOULD name all the important resources it exposes using CIDs. Determining which resources are
+  important is a matter of judgment, but anything that another agent might legitimately wish to
+  access is in scope, and it is best to err on the side of inclusion.
+- SHOULD expose the logical units of data that structure a resource (eg. a CBOR document, a file or
+  directory, a branch of a B-tree search index) using CIDs.
+- SHOULD support incremental verifiability, notably so that it may process content of arbitrary sizes.
+- MAY rely on any transport layer. The transport layer cannot dictate or constrain the way in which
+  CIDs map to content.
+
+## Boundary Examples
+
+These IPFS principles are broad. This is by design because, like HTTP, IPFS supports an open-ended set of
+use cases and is adaptable to a broad array of operating conditions. Considering cases
+at the boundary may help develop an intuition for the limits that these principles draw.
+
+### Other Content-Addressing Systems
+
+CIDs are readily made compatible with other content-addressable systems, but this does not
+entail that all content-addressable systems are part of IPFS. Git's SHA1 hashes aren't CIDs
+but can be converted into CIDs by prefixing them with `f01781114`. Likewise, BitTorrent v2
+uses multihashes in the `btmh:` scheme. BitTorrent addresses aren't CIDs, but can be
+converted to CIDs by replacing `btmh:` with `f017c`.
```

Comment:
```suggestion
converted to CIDs by replacing `btmh:` with `f017b`.
```

I know I'm the one who told you `c` but it was my mistake 😅. https://blog.libtorrent.org/2020/09/bittorrent-v2/ says `btmh` is for the info-hash and https://github.com/multiformats/multicodec/blob/6f018eb19666553c12955c24b8b5dc29fde6e9f4/table.csv#L48 is the one for info hashes (as described in https://github.com/ipld/js-ipld-torrent-file/pull/3#issuecomment-662137808, which granted is not a great source but unfortunately the code table doesn't always have obvious specs per code listed)

Reply:
LOL, I should have checked — good catch, thanks!